crates/re_types_builder/src/codegen/rust.rs

```diff
@@ -2411,8 +2411,7 @@ fn quote_arrow_field_deserializer(
                                 #[allow(unsafe_code, clippy::undocumented_unsafe_blocks)]
                                 let data = unsafe { data.get_unchecked(start as usize .. end as usize) };
 
-                                let mut arr = [Default::default(); #length];
-                                arr.copy_from_slice(data);
+                                let arr = array_init::from_iter(data.iter().copied()).unwrap(); // safe: checked above
```

Comment:
won't this fail if the `end - start != length` ? I don't see this being checked.

I guess it is guaranteed by how `offsets` is set up 30 lines above, but it is not obvious imho. Maybe this could be rewritten without the confusing `offsets` iterator, with just:

```rs
let starts = (0..).step_by(#length).take(#data_src.len());
starts.enumerate().map(move |(i, start)| {
    let end = start + #length;
    if end as usize > data.len() { 
    …
});
```

that would make it easier to see that `end - start == length`

Reply:
Same here; this will be part of the refactor PR.